File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.js
function solution(keyinput, board) {
    let x = 0;
    let y = 0;
    
    let xLimit = Math.floor(board[0]/2);
    let yLimit = Math.floor(board[1]/2);
    
    for(let i=0; i<keyinput.length; i++){
            if(keyinput[i] === "left" && -xLimit < x) x--;
            if(keyinput[i] === "right"&& xLimit > x) x++;
            if(keyinput[i] === "down" && -yLimit < y) y--;
            if(keyinput[i] === "up" && yLimit > y) y++;
    }
    return [x,y];
}

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.py
def solution(keyinput, board):
    answer = []
    mode = {
    "left":[-1,0],
    "right":[1,0],
    "down":[0,-1],
    "up":[0,1]}
    
    x,y = 0,0
    for key in keyinput:
        new_x, new_y = x + mode[key][0] , y + mode[key][1]
        if( -(board[0]//2) <= new_x <= board[0]//2 
           and -(board[1]//2) <= new_y <= board[1]//2):
            x,y = new_x , new_y
    return x,y